contracts/DollarToken.sol

```diff
@@ -86,10 +88,20 @@ contract DollarToken is Owned {
     {
         // TODO:
         //  receiveFee
-        //  allowWithdrawal
+        reserveFor(_beneficiaryAddr);
         Queue(queue).addBeneficiary(_beneficiaryAddr, msg.value);
     }
 
+    function reserveFor(address _participant)
+        payable
+    {
```

Comment:
I think it should work if you drop payable from privates / internal. I was doing this with a previous version of the EntryChannel and I'm pretty sure it was working : **https://github.com/Bakt/matcher/blob/411966197cfbda38f90ba73e91eef5b8db77ce68/contracts/BidChannel.sol**
.